blackboxopt/optimizers/testing.py

```diff
@@ -148,6 +150,70 @@ def is_deterministic_with_fixed_seed(
     return True
 
 
+def is_deterministic_when_reporting_shuffled_evaluations(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+) -> bool:
+    """Check if determinism isn't affected by the order of initially reported data.
+
+    Repeatedly initialize the optimizer with the same parameter space and a fixed seed.
+    Report a set of initial evaluations in randomized order as initial data. Start
+    optimizing and check if the generated configurations for all optimizers are equal.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
```

Comment:
Usually, tests fail on a problem but "quietly return"  when they pass. Is this considered "best practice" for reusable tests?

Reply:
Good question, @sfalkner, I just followed the pattern of the existing reference tests for consistency, but was also a bit confused by it. I'd suggest that someone removes those `return`s from all reference tests in separate PR.

Reply:
I'm responsible for that pattern but don't see a reason to abandon it.

Reply:
Removed all of those.